apix-impl/src/main/java/eu/xenit/apix/alfresco/sites/SiteService.java

```diff
@@ -43,26 +45,44 @@ public SiteService(ServiceRegistry serviceRegistry, ApixToAlfrescoConversion api
         List<SiteInfo> userSites = siteService.listSites(userId);
 
         for (SiteInfo userSite : userSites) {
-            NodeRef nodeRef = c.apix(userSite.getNodeRef());
-            String shortName = userSite.getShortName();
-            String title = userSite.getTitle();
-            String description = userSite.getDescription();
-            boolean isPublic = userSite.getIsPublic();
-            NodeRef documentLibrary = c.apix(siteService.getContainer(shortName, DOCUMENT_LIBRARY_COMPONENT));
-            NodeRef links = c.apix(siteService.getContainer(shortName, LINKS_COMPONENT));
-            NodeRef dataLists = c.apix(siteService.getContainer(shortName, DATA_LISTS_COMPONENT));
-            NodeRef wiki = c.apix(siteService.getContainer(shortName, WIKI_COMPONENT));
-            NodeRef discussions = c.apix(siteService.getContainer(shortName, DISCUSSIONS_COMPONENT));
-            Map<String, NodeRef> componentsMap = new HashMap<>();
-            componentsMap.put(DOCUMENT_LIBRARY_COMPONENT, documentLibrary);
-            componentsMap.put(LINKS_COMPONENT, links);
-            componentsMap.put(DATA_LISTS_COMPONENT, dataLists);
-            componentsMap.put(WIKI_COMPONENT, wiki);
-            componentsMap.put(DISCUSSIONS_COMPONENT, discussions);
 
-            apixSites.add(new Site(nodeRef, shortName, title, description, isPublic, componentsMap));
+                NodeRef nodeRef = c.apix(userSite.getNodeRef());
+                String shortName = userSite.getShortName();
+                String title = userSite.getTitle();
+                String description = userSite.getDescription();
+                boolean isPublic = userSite.getIsPublic();
+                Map<String, NodeRef> componentsMap;
```

Comment:
(non-blocking) If you want to avoid this declaration without assignment, you could also move the ` apixSites.add(..)` inside the try.